Display date format in Filter toolbar same as in Datagrid

nanjum30 opened the pull request:
Modifications done to use same date format as in DataGrid column for Display Date format in filter toolbar, if 'useDatagridFormat' property is set to true.
Property 'useDatagridFormat' can be set after initializing toolbar in onLoad by executing <toolbarFilter>.setDatagridFormat(true);
Note: By default, 'useDatagridFormat' property is set to false. Therefore, display date format will be same as previous i.e. "dd-MM-yyyy".
[This change Fixes # 16](https://github.com/Servoy/svyPopupFilter/issues/16)

nanjum30 committed:
Dispaly date format in Filter toolbar

Modifications done to use same date format as in DataGrid column for Display Date format in filter toolbar,if 'useDatagridFormat' property is set to true.
Property 'useDatagridFormat' can be set after initializing toolbar in onLoad by executing toolbarFIlter.setDatagridFormat(true);

paronne commented:
Hi Naveed,

thank you for your request and for your contribution. Before processing your request i would like to better understand the use case. 
The date format used to display the filter is now fixed, i agree would be nice to allow to configure the format. 

In your implementation you suggest to flag one option in the AbstractToolbarFilterUX object; i assume would be nice if all filters behave consistently, therefore what about a global setting such as "setConfigDateDisplayFormat(displayFormat)" for the toolbar filter instead of the flag configurable in the single AbstractToolbarFilterUX instance.

Also you suggest to use the same format of the grid, which can be an advantage since you don't nee to define the date format multiple times; do you actually expect to use different formatting per column ?
What if for some columns the date format shows only the month/year  "MM yyyy" ? the Date Filter always filter on specific dates e.g. ( 10/10/2020 - 20/10/2020); would be confusing if the filter shows format MM yyyy (10/2020 - 10/2020) as the column. 
Again would a global setting such as "setConfigDateDisplayFormat(displayFormat)", to be called once at the onSolutionOpen, be a definitive solution for your case ?

nanjum30 commented:
Hi Paolo,

Thanks for reviewing request and the changes. 

As suggested above, Global setting of display date format is really a nice approach. Just defining in 
OnSolutionOpen will do the work for all instances. 

Suggestion to use same format as in datagrid was to provide user with minimum input just a boolean flag and to avoid any wrong format input.

However, global configuration of display date format is a nice approach. Modifications for global configuration will be committed soon.

Regards,
Naveed

nanjum30 committed:
Global date display format in Filter toolbar

Global date display format can be configured by function "setConfigDateDisplayFormat(displayFormat)". This needs to be called once at the SolutionOpen. By default, date format will be "dd-MM-yyyy" as previous.

paronne commented:
Hi Naveed, thank you for updating your pull request and for your great contribution.
